pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,154 @@
+{ lib
+, pkgs
+, stdenv
+, writeScript
+, writeShellScript
+, nix
+, gradle
+, openssl
+, mitm-cache
+, jdk
+, nsjail
+, path
+, python3
+}:
+
+# path to the derivation attr (usually the package name)
+{ attrPath ? pname
+# pname (attrPath synonym)
+, pname ? ""
+# the Nix code to get the derivation (this will be put in Bash double quotes)
+# this is relevant for downstream (non-nixpkgs) users
+, code ? "with import ./. { }; ($attrPath)"
+# unlike gradleFlags in normal packages, these flags are escaped so they are allowed to contain spaces
+# also, gradleFlags from configurePhase are used by default
+, gradleFlags ? []
+# attribute overrides for the "source" derivation
+, attrOverrides ? {}
+# run before fetching the dependencies (in sandbox)
+, preBuild ? ""
+# run before starting the sandbox for fetching dependencies
+, preBuildNoSandbox ? ""
+# the gradle commands for fetching the necessary dependencies
+, gradleCommands ? "gradle nixDownloadDeps"
+# run after fetching the dependencies (in sandbox)
+, postBuild ? ""
+# run after the sandbox for fetching dependencies has stopped
+, postBuildNoSandbox ? ""
+, mitmCachePort ? 1337
+# deps path (relative to the package directory), or absolute
+, depsPath ? "deps.json"
+# extra packages to make available in the sandbox
+, availablePackages ? [ ]
+}:
+
+let
+  keep = [ "MITM_CACHE_PORT" "MITM_CACHE_KEYSTORE" "MITM_CACHE_KS_PWD" "MITM_CACHE_CA" ];
+  availablePackages' = map (drv: ''(builtins.storePath "${drv}")'') availablePackages;
+  gradleScript = writeScript "gradle-commands.sh" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell --pure ${lib.concatMapStringsSep " " (x: "--keep ${x}") keep} -i bash -p coreutils ${lib.concatStringsSep " " availablePackages'}
```

Comment:
Could this script pin `bash`, `coreutils`, etc to make sure it behaves consistently across platforms. `nix-shell` may also not be available on some machines.